modules/network/src/main/java/org/apache/ignite/internal/network/netty/OutboundEncoder.java

```diff
@@ -136,8 +138,8 @@ public ByteBuf readChunk(ChannelHandlerContext ctx) throws Exception {
 
         /** {@inheritDoc} */
         @Override
-        public ByteBuf readChunk(ByteBufAllocator allocator) throws Exception {
-            ByteBuf buffer = allocator.ioBuffer();
+        public ByteBuf readChunk(ByteBufAllocator allocator) {
+            ByteBuf buffer = allocator.ioBuffer(IO_BUFFER_CAPACITY);
             int capacity = buffer.capacity();
 
             ByteBuffer byteBuffer = buffer.internalNioBuffer(0, capacity);
```

Comment:
Interestingly, I see "internal use only" in documentation to this method))

Reply:
The method javadoc follows:

```
    /**
     * Allocate a {@link ByteBuf}, preferably a direct buffer which is suitable for I/O.
     */
```

Where do you see that 'internal use only'?

Reply:
Oh, it's about `internalNioBuffer()`. We don't use the resizability of `ByteBuf`; instead, we use the original NIO buffer, for speed.